server/leafnode.go

```diff
@@ -888,7 +888,13 @@ func (s *Server) createLeafNode(conn net.Conn, rURL *url.URL, remote *leafNodeCf
 	}
 	now := time.Now().UTC()
 
-	c := &client{srv: s, nc: conn, kind: LEAF, opts: defaultOpts, mpay: maxPay, msubs: maxSubs, start: now, last: now}
+	var clientOpts = ClientOpts{
+		Name:     opts.ServerName,
```

Comment:
This is the only bit I'm not feeling 100% about. For outgoing connections I set the Name as the current server name. I believe this is probably what we want, since folks looking at LEAFZ should be able to tell the difference between incoming and outgoing connections here, but would like some feedback

Reply:
LMK if you want me to merge it.

Reply:
Go for it!